lib/db/CountLocationDAO.js

```diff
@@ -0,0 +1,110 @@
+import { CardinalDirection } from '@/lib/Constants';
+import db from '@/lib/db/db';
+import ArteryDAO from '@/lib/db/ArteryDAO';
+import CentrelineDAO from '@/lib/db/CentrelineDAO';
+import CountLocation from '@/lib/model/CountLocation';
+
+const COUNT_LOCATION_FIELDS = `
+  FALSE AS "legacy",
+  cl."id",
+  cl."description",
+  cl."centrelineId",
+  cl."centrelineType",
+  ST_AsGeoJSON(cl."geom")::json AS "geom"
+  FROM counts2.count_locations cl`;
+
+const REGEX_DIRECTION_BOUND = / [NESW]\/B/;
+
+async function validateCountLocation(countLocation) {
+  if (countLocation === null) {
+    return null;
+  }
+  return CountLocation.read.validateAsync(countLocation);
+}
+
+/**
+ * Data access object for count locations, i.e. locations at which counts have been conducted.
+ * This is used in the new `counts2` schema.
+ *
+ * To help transition from the now-deprecated {@link ArteryDAO} to this, we fetch both legacy
+ * arterycodes and new `counts2.count_locations` records here, adapting the former into the same
+ * data model as the latter.
+ */
+class CountLocationDAO {
+  /**
```

Comment:
we don't need this method - `direction` is a property of _counts_ now, not of _locations_.

Reply:
added this to #971